### ISSUE TITLE
Periodic, inevitable lag spikes caused by Optifine

### ISSUE THREAD

**Issue by dhasial** [GTNewHorizons/GT-New-Horizons-Modpack#12027]
### Your Pack Version

2.2.7

### Your Server

SP

### Type of Server

Single Player

### Your Expectation

I did not expect lag spikes to make the pack unplayable. This issue is unique to this pack.

*This was originally another issue, #11910 which was closed after another used commented that they supposedly had the same issue, and then resolved it. I've since found a bit more information than I had originally, and have included logs from both the launcher and visualvm

### The Reality

At a seemingly random interval, FPS drop from a range of 50-60 to 1-7 until I press F3 + A to reload all chunks. After doing so everything goes back to normal until the lag inevitably returns. Rinse and repeat. I made a new world wondering if it was a corrupted chunk or something, but no, within 3 minutes FPS dropped to unplayable levels until I pressed F3 + A again.

This is on an SSD, with autosaves set to every 30 minutes in case that's what causes it (I don't believe it is since the lag happens more frequently than that). I also disabled AromaBackup at one point to narrow it down, didn't change anything.

*Everything else in this section is new information/new logs

I've since found that this issue seems to only occur when Optifine is being used. This occurs both with/without shaders being used. It also occurs with both versions of Optifine tested, "1.7.10 HD U D6", and moreso "1.7.10 HD U E7". I've also tried using Liberica, OpenJ9, and normal Oracle versions of Java, none of which seem to affect the bug.

OS: Arch Linux
Window manager: Sway (Wayland)

Polymc log: 
[GTNH-2.2.7-bug-2022-12-18T13.37.46-0500.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/10254574/GTNH-2.2.7-bug-2022-12-18T13.37.46-0500.log)

**The following files are the heap dump, thread dump, and application snapshot all generated from visualvm. Because of the size of the uncompressed heap dump file (3,7GB), and because of Github's issue-file-uploader feature not allowing that size+all the file extensions, I had to zip all of the files together, and upload them to mega.nz

https://mega.nz/file/33hnWBRK#ixW9gXE3dsYrjAHmU6OWaQh488ll88hzDGurPpoFEZY

### Your Proposal

I'm honestly not sure where to start.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by chochem** [comment]
A fixes has been made since 2.2.7 that resolved lag spikes for many people. You can try that out with the latest gregtech from https://github.com/GTNewHorizons/GT5-Unofficial/releases

However, afaik there was nothing optifine related in that. So it might not help with your issue.

**Comment by dhasial** [comment]
So I tried the latest release out (5.09.41.180-pre), but unfortunately the seemingly optifine-related lag spikes are still occurring.

**Comment by dhasial** [comment]
I tried the latest release out (5.09.41.180-pre), but unfortunately the seemingly optifine-related lag spikes are still occurring.

**Comment by chochem** [comment]
Ah well, was worth a try. Have you tried playing around a bit with the optifine settings? Might be a specific thing.

**Comment by dhasial** [comment]
I have fiddled with the setting for quite a while and sadly I can't find anything that seems to affect it (though it's possible I might've missed an option).

**Comment by dhasial** [comment]
I've discovered that the issue was due to a very niche configuration of mine.

Just on the astronomically small chance that this affects someone else - 
My issue was due to using the CPU scheduler TT (TaskType) in my kernel. It took me a month+ to do my occasional kernel rebuild and I just happened to want to try out another scheduler (BORE scheduler) which completely resolved not only the lag spikes, but nigh all lag within GTNH. Strangely this interaction only occurred when using both TaskType, GTNH AND Optifine, other modpacks were unaffected (FTB Revelations on 1.12.2, All The Fabric 6 on 1.19, and a tiny bit of Enigmatica 9). Perhaps my desktop performance seems to be slightly better, but it's nothing remotely as drastic as it is on GTNH.

Anyways, it was completely on my end, it runs like a dream with shaders now, thanks all you folks for making this absolute beaut' of a modpack.
